src/Scope.zig

```diff
@@ -248,6 +309,67 @@ pub fn findBlockReturnType(inner: *Scope) aro.QualType {
     }
 }
 
+pub fn processContainerMemberFnsList(t: *Translator, list: std.ArrayListUnmanaged(ContainerMemberFns)) !void {
+    for (list.items) |item| {
+        const decl_tag = item.container_decl.tag();
+        const decl_data = switch (decl_tag) {
+            inline .var_simple, .pub_var_simple => |tag| item.container_decl.castTag(tag).?.data,
+            else => unreachable,
+        };
+
+        var container_data = switch (decl_data.init.tag()) {
+            inline .@"struct", .@"union" => |tag| &decl_data.init.castTag(tag).?.data,
+            else => unreachable,
+        };
+
+        const arena = t.arena;
+        const old_variables = container_data.variables;
+        var new_variables = try arena.alloc(ast.Node, old_variables.len + item.member_fns.items.len);
+        @memcpy(new_variables[0..old_variables.len], old_variables);
+        // Assume the allocator of container_data.variables is arena,
+        // so don't add arena.free(old_variables).
+        var func_ref_vars = new_variables[old_variables.len..];
+        var last_name_array = try t.gpa.alloc([]const u8, item.member_fns.items.len);
+        defer t.gpa.free(last_name_array);
+        var count: u32 = 0;
+        for (item.member_fns.items, 0..) |func, i| {
+            const func_name = func.data.name orelse continue;
+            const ident_payload = try t.arena.create(ast.Payload.Value);
+            ident_payload.* = .{
+                .base = .{ .tag = .identifier },
+                .data = func_name,
+            };
+
+            const last_index = std.mem.lastIndexOf(u8, func_name, "_");
+            const last_name = if (last_index) |index| func_name[index + 1 ..] else continue;
+            last_name_array[i] = last_name;
+            var same_count: u32 = 0;
+            for (last_name_array[0..i]) |name| {
+                if (std.mem.eql(u8, name, last_name)) {
+                    same_count += 1;
+                }
+            }
+            const var_name = if (same_count == 0)
+                last_name
+            else
+                try std.fmt.allocPrint(t.arena, "{s}{d}", .{ last_name, same_count });
+
+            // compare all before last_name
+            const payload = try t.arena.create(ast.Payload.SimpleVarDecl);
+            payload.* = .{
+                .base = .{ .tag = decl_tag },
```

Comment:
These should always be pub, there is no reason to add non-pub aliases since you wouldn't be able to use them.